apps/taskman/service.py

```diff
@@ -0,0 +1,180 @@
+        """get Jira task given its string key or integer id"""
+        try:
+            return Response(data=self.jira_conn.issue(task_key).raw, status=200)
+        except JIRAError as e:
+            return Response(data=e.response.json(), status=e.status_code)
+
+    def create_or_update_task(self, flaw: Flaw, fail_if_exists=False) -> Response:
+        """Creates a task using Flaw data"""
+        task = self.get_task_by_flaw(flaw_uuid=flaw.uuid).data
+        if fail_if_exists and task:
+            res = {
+                "error": "Task representing this flaw already exists",
+                "existing_task": task,
+            }
+            return Response(data=res, status=409)
+
+        try:
+            if task:
+                data = {
+                    "fields": {
+                        "summary": flaw.title,
+                        "description": flaw.description,
+                    }
+                }
+
+                url = f"{self.jira_conn._get_url('issue')}/{task['key']}"
+                r = self.jira_conn._session.put(url, json.dumps(data))
+                return Response(status=r.status_code)
+            else:
+                data = {
+                    "fields": {
+                        "issuetype": {
+                            "id": self.jira_conn.issue_type_by_name("Story").id
+                        },
+                        "project": {
+                            "id": self.jira_conn.project(
+                                get_env("JIRA_TASKMAN_PROJECT_KEY")
+                            ).id
+                        },
+                        "summary": flaw.title,
+                        "description": flaw.description,
+                        "labels": [f"flawuuid:{flaw.uuid.__str__()}"],
+                    }
+                }
+
+                issue = self.jira_conn.create_issue(
+                    fields=data["fields"], prefetch=True
+                )
+                return Response(data=issue.raw, status=201)
+        except JIRAError as e:
+            return Response(data=e.response.json(), status=e.status_code)
+
+    def update_task_status(self, issue_key: str, status: TaskStatus) -> Response:
+        """Transition a task to a new state"""
+        try:
+            self.jira_conn.transition_issue(issue=issue_key, transition=status)
+            return Response(data={}, status=200)
+        except JIRAError as e:
+            return Response(data=e.response.json(), status=e.status_code)
+
+    def create_comment(self, issue_key: str, user: str, body: str):
+        """Add a comment in a task"""
+        try:
+            comment = self.jira_conn.add_comment(issue_key, f"{user}: {body}")
+            return Response(data=comment.raw, status=201)
+        except JIRAError as e:
+            return Response(data=e.response.json(), status=e.status_code)
+
+    def update_comment(self, issue_key, comment_id, user: str, body: str):
```

Comment:
The comment seem to me as a thing naturally requiring the user audit (instead of just prefixes). This should be used for some internal communication and I do not think it would be correct to have these comments created/updated by the service account. The comment actions are direct (the task tracking actions might otherwise be indirect) and really performed by the user.